adapters/repos/db/vector/hnsw/commit_logger.go

```diff
@@ -441,48 +457,58 @@ func (l *hnswCommitLogger) RootPath() string {
 	return l.rootPath
 }
 
-func (l *hnswCommitLogger) startSwitchLogs(stopFunc cyclemanager.StopFunc) {
-	if err := l.SwitchCommitLogs(false); err != nil {
+func (l *hnswCommitLogger) startSwitchLogs(shouldBreak cyclemanager.ShouldBreakFunc) bool {
+	err, executed := l.switchCommitLogs(false)
+	if err != nil {
 		l.logger.WithError(err).
 			WithField("action", "hnsw_commit_log_maintenance").
 			Error("hnsw commit log maintenance failed")
 	}
+	return executed
 }
 
-func (l *hnswCommitLogger) startCombineAndCondenseLogs(stopFunc cyclemanager.StopFunc) {
-	if err := l.combineLogs(); err != nil {
+func (l *hnswCommitLogger) startCombineAndCondenseLogs(shouldBreak cyclemanager.ShouldBreakFunc) bool {
+	err, executed1 := l.combineLogs()
```

Comment:
I think it might be cleaner to make sure that the error is always the last arg. E.g.:

```
ok, err := something()
```

is more expected than

```
err, ok := something()
```

Reply:
fixed